### PR TITLE
fix: makes default token_url universe aware

### PR DESCRIPTION
**Pull request by aeitzman**
Also adds tests for credentials that use the default token_url

### CODE REVIEW

File: google/auth/external_account.py [hunk 1]
```diff
@@ -147,7 +147,12 @@ def __init__(
         super(Credentials, self).__init__()
         self._audience = audience
         self._subject_token_type = subject_token_type
+        self._universe_domain = universe_domain or credentials.DEFAULT_UNIVERSE_DOMAIN
```
Comment (clundin25) [inline google/auth/external_account.py:150]: Is this already available from the parent class?

File: google/auth/external_account.py [hunk 2]
```diff
@@ -52,7 +52,7 @@
 # Cloud resource manager URL used to retrieve project information.
 _CLOUD_RESOURCE_MANAGER = "https://cloudresourcemanager.googleapis.com/v1/projects/"
 # Default Google sts token url.
-_DEFAULT_TOKEN_URL = "https://sts.googleapis.com/v1/token"
+_DEFAULT_TOKEN_URL = "https://sts.{universe_domain}/v1/token"
```
Comment (clundin25) [inline google/auth/external_account.py:55]: This is fine for now but if the URL continues to get complex it should be wrapped in a type.

**Review by clundin25 — APPROVED** [review]

File: google/auth/external_account.py (hunk 2 above)
Comment (aeitzman) [reply google/auth/external_account.py:55]: SG, should not get more complex than this, but will make that change if it does.

File: google/auth/external_account.py (hunk 1 above)
Comment (aeitzman) [reply google/auth/external_account.py]: The defaulting is, but it doesn't set the universe_domain from the constructor.

**Pull request merged**